addons/sourcemod/scripting/nd_bot_radio_helpers.sp

```diff
@@ -0,0 +1,455 @@
+#include <sourcemod>
+#include <sdktools>
+#include <sdkhooks>
+#include <nd_ammo>
+#include <nd_classes>
+
+#define PLUGIN_VERSION "1.0.1"
+
+public Plugin myinfo =
+{
+    name = "[ND] Bot Medic/Engineer Radio Helper",
+    author = "databomb",
+    description = "Adds bot ability to drop medkits/ammopacks when players radio for support",
+    version = PLUGIN_VERSION,
+    url = "https://forums.alliedmods.net/member.php?u=54648"
+};
+
+bool g_bMedpackThrown[MAXPLAYERS+1];
+bool g_bAmmopackThrown[MAXPLAYERS+1];
+int g_iAmmopackRequested[MAXPLAYERS+1];
+int  g_iMedpackRequested[MAXPLAYERS+1];
+float g_fEquipTime[MAXPLAYERS+1];
+float g_fFollowTime[MAXPLAYERS+1];
+
+public OnPluginStart()
+{
+    UserMsg hSayTextTwo = GetUserMessageId("SayText2");
+    if (hSayTextTwo == INVALID_MESSAGE_ID)
+    {
+        SetFailState("Failed to find user message SayText2");
+    }
+
+    HookUserMessage(hSayTextTwo, MessageHook_SayTextTwo);
+
+    // Account for late loading
+    for (int iClient = 1; iClient <= MaxClients ; iClient++)
+    {
+        if (IsClientInGame(iClient) && IsFakeClient(iClient))
+        {
+            SDKHook(iClient, SDKHook_WeaponCanSwitchTo, OnWeaponCanSwitchTo);
+        }
+    }
+}
+
+public OnClientPutInServer(client)
+{
+    if (IsFakeClient(client))
+    {
+        SDKHook(client, SDKHook_WeaponCanSwitchTo, OnWeaponCanSwitchTo);
+    }
+}
+
+public Action OnWeaponCanSwitchTo(client, weapon)
+{
+    eNDClass ePlayerClass = ND_GetPlayerClass(client);
+
+    switch (ePlayerClass)
+    {
+        case eNDClass_SupportMedic:
+        {
+            if (g_iMedpackRequested[client])
+            {
+                char sWeaponName[32];
+                GetEdictClassname(weapon, sWeaponName, sizeof(sWeaponName));
+                if (!StrEqual(sWeaponName, "weapon_medpack"))
+                {
+                    return Plugin_Handled;
+                }
+            }
+        }
+        case eNDClass_SupportEngineer:
+        {
+            if (g_iAmmopackRequested[client])
+            {
+                char sWeaponName[32];
+                GetEdictClassname(weapon, sWeaponName, sizeof(sWeaponName));
+                if (!StrEqual(sWeaponName, "weapon_ammopack"))
+                {
+                    return Plugin_Handled;
+                }
+            }
+        }
+    }
+
+    return Plugin_Continue;
+}
+
+public Action OnPlayerRunCmd(int client, int &buttons, int &impulse, float vel[3], float angles[3])
+{
+    if(client && IsClientInGame(client) && IsFakeClient(client) && IsPlayerAlive(client))
+    {
+        eNDClass ePlayerClass = ND_GetPlayerClass(client);
+        char sClassname[64];
+
+        switch (ePlayerClass)
+        {
+            case eNDClass_SupportMedic:
+            {
+                if (g_iMedpackRequested[client])
```

Comment:
Could we get the player class when the bot spawns? It already assigned for accordingly to `g_iMedpackRequested[client]` and `g_iAmmopackRequested[client]`. 

We just need a boolean `g_isSupportMedic[client]` and `g_isSupportEngineer[client]`.

The code is as follows:
```
public void OnPluginStart()
{
      HookEvent("player_spawn", PlayerSpawn);
}
public Action PlayerSpawn(Event event, const char[] name, bool dontBroadcast)
{
      // Do stuff here
}
```

The reason for this requested change is to avoid calling `GetEntProp` frequently during `OnPlayerRunCmd`.

Reply:
Reduces the calls to GetEntProp during OnPlayerRunCmd https://github.com/data-bomb/Redstone/commit/59371611dfb10531516012bc613a0b6a1426456e